mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -244,17 +248,33 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
                 env=parallel_download_subproc_env,
                 headers=self._extract_headers_from_credentials(cloud_credential_info.headers),
             )
-
-            if failed_downloads:
+            num_retries = _MLFLOW_MPD_NUM_RETRIES.get()
+            interval = _MLFLOW_MPD_RETRY_INTERVAL_SECONDS.get()
+            failed_downloads = list(failed_downloads)
+            while failed_downloads and num_retries > 0:
```

Comment:
what happens when num_retries == 0, and downloads are still failing for some reason? 
looks like this code will succeed and we end up with incomplete model artifacts unknowingly?

Reply:
lmk if the assert statement is a valid way to handle this. Looks like its been done with read_credentials as well, so just following suit

Reply:
if retry count goes to 0 and we still have a non empty failed_downloads, this should raise an exception

> Looks like its been done with read_credentials as well
What's read_credentials? code pointer?

Reply:
sorry, i think the comment got sent without me pushing the changes. I was thinking of just doing an assert statement (assert num_retries > 0) similar to how its done in https://github.com/mlflow/mlflow/blob/a1b9c0e01aa438fa49e0ee5d7f450009be6a281d/mlflow/store/artifact/cloud_artifact_repo.py#L231. However, after some thought, I just went forward and threw an error using an if statement. Let me know if you'd prefer that I assert instead of throwing an error.

Reply:
yea we should throw an exception with error message that says this download still fails after x retry attempts.

Also, `assert num_retries > 0` may not be correct? e.g what if when the loop completes, num_retries == 0 and all the chunks have been downloaded successfully.

The condition to throw error / assert should be 1) num_retries == 0  && 2) failed_downloads is not empty

Reply:
got it, should be fixed now